lib/services/watcher.go

```diff
@@ -880,3 +880,157 @@ func appsToSlice(apps map[string]types.Application) (slice []types.Application)
 	}
 	return slice
 }
+
+// CertAuthorityWatcherConfig is a CertAuthorityWatcher configuration.
+type CertAuthorityWatcherConfig struct {
+	// ResourceWatcherConfig is the resource watcher configuration.
+	ResourceWatcherConfig
+	// AuthorityGetter is responsible for fetching cert authority resources.
+	AuthorityGetter
+	// CertAuthorityC receives up-to-date list of all cert authority resources.
+	CertAuthorityC chan []types.CertAuthority
```

Comment:
any reason this `chan` is owned by the config not the watcher itself?

Reply:
I mimicked the pattern the existing resource watchers used. This is mostly used for testing purposes.

Reply:
ok. now i see other xxxConfig is doing this. To me at least it feels weird that a Config *owns* a chan, and especially no caller (non-test ocde) is going to set it. For tests, i am sure there are many ways to make it happen. Anyway lets see if others have comment on this. I will approve for now.